File: Modeling .NET/VB/Queens/Queens/QueensRun.vb
Imports Simulation

Public Class QueensRun
    Inherits Coroutine

    Protected Overrides Sub Run()
        Dim i As Byte
        For i = 0 To Board.QueenCount - 1
            Board.Queens(i) = New Queen(i)
        Next
        SwitchTo(Board.Queens(0))
        For i = 0 To Board.QueenCount - 1
            Board.Queens(i).Finish()
        Next
        Detach()
    End Sub
End Class

File: Modeling .NET/Delphi/Queens/QueensRun.pas
unit QueensRun;

interface
uses Simulation;

type
  TQueensRun = class(Coroutine)
  strict protected
    procedure Run; override;
  end;

implementation
uses Queen, Board;

{ TQueensRun }

procedure TQueensRun.Run;
var
  i : Byte;
begin
  inherited;
  for i := 0 to QueensCount - 1 do
    TBoard.Queens[i] := TQueen.Create(i);
  TBoard.Queens[0].SwitchTo;
  for i := 0 to QueensCount - 1 do
    TBoard.Queens[i].Finish;
end;

end.
